owncloudApp/src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java

```diff
@@ -99,8 +99,12 @@ protected void onCreate(Bundle savedInstanceState) {
                 PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(getApplicationContext())
         );
 
-        setupToolbar();
-        updateActionBarTitleAndHomeButtonByString(getResources().getString(R.string.prefs_manage_accounts));
+        ToolbarConfig.ToolbarStandard toolbarConfig = new ToolbarConfig.ToolbarStandard(
+                getResources().getString(R.string.prefs_manage_accounts),
+                true,
+                true
+        );
+        setupToolbar(toolbarConfig);
```

Comment:
Instead of setting up the toolbar in so many activities, why don't you create some parent activities like `ToolbarStandardActivity`? Is it possible?

Reply:
All of them extends from ToolbarActivity. The problem here is that we need to set a title and if we need to set the back arrow. I tried to keep a similar behavior, but I considered some different options too. For example, create a new component OwncloudToolbar, and set in the xml the title, back navigation, etc using attrs. But it was too complex.

Reply:
Ok, I've asked this in order to avoid some logic in activities like https://github.com/owncloud/android/pull/3087/files#diff-0d32bb6cd423d0567750a9f31037db8e2edadc51b3f583685d85a3aa865d37fcR1132 . Maybe creating something like

```
class Toolbar(
    val title: String?,
    val displayHomeAsUpEnabled: Boolean,
    val homeButtonEnabled: Boolean,
    val enableSearch: Boolean
)
```

And a

```
class ToolbarCreator() {
    fun getToolbar(
        remotePath: String
    ): Toolbar {
        if (remotePath == "whatever"...) {
              return Toolbar (whatever)
        }
    }
}
```

From a viewmodel calls:

toolbarCreator.getToolbar( )

And you set the toolbar to the fragment via a liveData so the fragment doesn't know anything about the logic.

It's just an idea if you want to do something like that in the future.

Reply:
Hard to do at the moment. Let's wait a little bit more till the new arch is finished.